pkg/actors/actor.go

```diff
@@ -53,18 +53,24 @@ type actor struct {
 	// is used when runtime drains actor.
 	disposeCh chan struct{}
 
-	once sync.Once
+	once  sync.Once
+	clock clocklib.Clock
 }
 
-func newActor(actorType, actorID string, maxReentrancyDepth *int) *actor {
+func newActor(actorType, actorID string, maxReentrancyDepth *int, clock clocklib.Clock) *actor {
```

Comment:
nit: If you use another ctor method `newActorWithClock`, you can avoid making so many `newActor(..., nil)` calls.